firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,491 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import re
+
+import mozinfo
+
+from ..base import BaseLib
+from .appinfo import AppInfo
+from .prefs import Preferences
+
+
+class SoftwareUpdate(BaseLib):
+    """The SoftwareUpdate API adds support for an easy access to the update process."""
+
+    PREF_APP_DISTRIBUTION = 'distribution.id'
+    PREF_APP_DISTRIBUTION_VERSION = 'distribution.version'
+    PREF_APP_UPDATE_URL = 'app.update.url'
+    PREF_DISABLED_ADDONS = 'extensions.disabledAddons'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.app_info = AppInfo(marionette_getter)
+        self.prefs = Preferences(marionette_getter)
+
+        self.update_channel = UpdateChannel(marionette_getter)
+        self.mar_channels = MARChannels(marionette_getter)
+        self.active_update = ActiveUpdate(marionette_getter)
```

Comment:
I could add properties for them, but that seems a bit redundant. Do you know how to flag them to be picked up for the docs, or is that something you want me to figure out?